aries_cloudagent/config/wallet.py

```diff
@@ -79,4 +79,7 @@ async def wallet_config(context: InjectionContext, provision: bool = False):
             seed=test_seed, metadata={"endpoint": "1.2.3.4:8021"}
         )
 
+    if public_did:
+        context.settings["wallet.public_did"] = public_did
```

Comment:
There's a method on the wallet called `get_public_did` which does a dynamic lookup for this, because there are admin endpoints to dynamically change the public DID. If it's stored in the context then there isn't a reliable way to update it everywhere